shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/distsql/ral/common/set/excutor/SetVariableExecutor.java

```diff
@@ -41,20 +51,57 @@
     
     @Override
     public ResponseHeader execute() {
-        SetVariableStatement setVariableStatement = sqlStatement;
-        VariableEnum variable = VariableEnum.getValueOf(setVariableStatement.getName());
+        Enum type = getType(sqlStatement.getName());
+        if (type instanceof ConfigurationPropertyKey) {
+            handleConfigurationProperty((ConfigurationPropertyKey) type, sqlStatement.getValue());
+        } else if (type instanceof VariableEnum) {
+            handleVariables(sqlStatement);
+        } else {
+            throw new UnsupportedVariableException(sqlStatement.getName());
+        }
+        return new UpdateResponseHeader(sqlStatement);
+    }
+    
+    private Enum getType(final String name) {
+        try {
+            return ConfigurationPropertyKey.valueOf(name.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            return VariableEnum.getValueOf(name);
+        }
+    }
+    
+    private void handleConfigurationProperty(final ConfigurationPropertyKey propertyKey, final String value) {
+        MetaDataContexts metaDataContexts = ProxyContext.getInstance().getContextManager().getMetaDataContexts();
+        Optional<MetaDataPersistService> metaDataPersistService = metaDataContexts.getMetaDataPersistService();
+        Properties props = metaDataContexts.getProps().getProps();
+        props.put(propertyKey.getKey(), getValue(propertyKey, value));
+        if (metaDataPersistService.isPresent() && null != metaDataPersistService.get().getPropsService()) {
+            metaDataPersistService.get().getPropsService().persist(props, true);
+        }
+    }
+    
+    private Object getValue(final ConfigurationPropertyKey propertyKey, final String value) {
+        try {
+            TypedPropertyValue propertyValue = new TypedPropertyValue(propertyKey, value);
+            return propertyValue.getValue();
+        } catch (TypedPropertyValueException e) {
+            throw new InvalidValueException(value);
+        }
+    }
+    
+    private void handleVariables(final SetVariableStatement statement) {
```

Comment:
The variable name `statement` represents a lot of meaning, can it be more precise?